aten/src/ATen/cuda/tunable/TunableOp.h

```diff
@@ -120,31 +123,62 @@ class TunableOp {
       auto min_duration_ms = std::numeric_limits<double>::infinity();
       std::string id_name = "Default";
 
+      int flush_iters = ctx->GetICacheFlushIterations();
+      if (flush_iters > 0) {
+        TUNABLE_LOG("instruction cache flush is enabled");
+      }
+      for (int i = 0; i < flush_iters; i++) {
+        at::cuda::flush_icache();
+      }
+
       // calcaulte a reference answer for numerical check
-      ParamsT* reference_params = params->DeepCopy();
+      ParamsT* reference_params = params->DeepCopy(false);
       TORCH_CHECK(ops_[ResultEntry::Default()]->Call(reference_params) == OK);
 
-      // need a copy of params to reuse
-      ParamsT* reusable_params = params->DeepCopy();
+      // need copies of params to reuse
+      // make as many copies as will fill the requested rotating buffer size, if requested
+      size_t rotating_size = ctx->GetRotatingBufferSize();
+      bool use_buffer_rotation = (rotating_size > 0);
+      size_t param_size = params->GetSize(use_buffer_rotation);
+      size_t param_count = (rotating_size / param_size) + 1;
+      if (use_buffer_rotation) {
+        TUNABLE_LOG("Rotating buffer ", rotating_size/(1024*1024), " MiB. ",
+            "Needed Size: ", param_size/(1024*1024), " MiB. ",
```

Comment:
minor: define a constant of MB=1024*1024?

Reply:
Done as suggested.